ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LedgerPeers/Type.hs

```diff
@@ -15,17 +22,137 @@ module Ouroboros.Network.PeerSelection.LedgerPeers.Type
   , UseLedgerPeers (..)
   , AfterSlot (..)
   , LedgerPeersKind (..)
+  , LedgerPeerSnapshot (.., LedgerPeerSnapshot)
   , isLedgerPeersEnabled
   ) where
 
-import Cardano.Slotting.Slot (SlotNo (..), WithOrigin)
+import Control.Monad (forM)
+import Data.List.NonEmpty (NonEmpty)
+import GHC.Generics (Generic)
+import Text.Read (readMaybe)
+
+import Cardano.Binary (FromCBOR (..), ToCBOR (..))
+import Cardano.Binary qualified as Codec
+import Cardano.Slotting.Slot (SlotNo (..), WithOrigin (..))
 import Control.Concurrent.Class.MonadSTM
 import Control.DeepSeq (NFData (..))
-import Data.List.NonEmpty (NonEmpty)
-import GHC.Generics
+import Data.Aeson
+import Data.Aeson.Types
 import NoThunks.Class
 import Ouroboros.Network.PeerSelection.RelayAccessPoint (RelayAccessPoint)
 
+-- |The type of big ledger peers that is serialised or later
+-- provided by node configuration for the networking layer
+-- to connect to when syncing.
+--
+data LedgerPeerSnapshot =
+  LedgerPeerSnapshotV1 (WithOrigin SlotNo, [(AccPoolStake, (PoolStake, NonEmpty RelayAccessPoint))])
+  -- ^ Internal use for version 1, use pattern synonym for public API
+  deriving (Eq, Show)
+
+-- |Public API to access snapshot data. Currently access to only most recent version is available.
+-- Nonetheless, serialisation from the node into JSON is supported for older versions via internal
+-- api so that newer CLI can still support older node formats.
+--
+pattern LedgerPeerSnapshot :: (WithOrigin SlotNo, [(AccPoolStake, (PoolStake, NonEmpty RelayAccessPoint))])
+                           -> LedgerPeerSnapshot
+pattern LedgerPeerSnapshot payload <- LedgerPeerSnapshotV1 payload where
+  LedgerPeerSnapshot payload = LedgerPeerSnapshotV1 payload
+
+{-# COMPLETE LedgerPeerSnapshot #-}
+
+-- | In case the format changes in the future, this function provides a migration functionality
+-- when possible.
+--
+migrateLedgerPeerSnapshot :: LedgerPeerSnapshot
+                          -> Maybe (WithOrigin SlotNo, [(AccPoolStake, (PoolStake, NonEmpty RelayAccessPoint))])
+migrateLedgerPeerSnapshot (LedgerPeerSnapshotV1 lps) = Just lps
+
+instance ToJSON LedgerPeerSnapshot where
```

Comment:
Do we have round trip tests for `{To,From}JSON` instances?

Reply:
Yes.

Reply:
Yes, see above.